rerun_cpp/src/rerun/config.hpp

```diff
@@ -0,0 +1,45 @@
+#pragma once
+#include <atomic>
+
+#ifndef RERUN_ENABLED
+#define RERUN_ENABLED 1
+#endif
+
+namespace rerun {
+    struct RerunGlobalConfig {
+        static RerunGlobalConfig& instance();
+
+        RerunGlobalConfig(const RerunGlobalConfig&) = delete;
+        RerunGlobalConfig& operator=(const RerunGlobalConfig&) = delete;
+
+        std::atomic_bool default_enabled;
+
+      private:
+        RerunGlobalConfig();
+
+        ~RerunGlobalConfig() {}
+    };
+
+    /// Change whether `RecordingStream`s are enabled by default.
+    ///
+    /// This governs the creation of new `RecordingStream`s. If `default_enabled` is
+    /// is `false`, `RecordingStreams` will be created in the disabled state. Changing
+    /// the value of `default_enabled` will not affect existing `RecordingStream`s.
+    ///
+    /// Note that regardless of usage of this API, the value of default_enabled will
+    /// be overridden by the RERUN environment variable.
+    ///
+    /// If RERUN is set to `1`, `true`, or `yes`, then Rerun is enabled. If RERUN is
+    /// set to `0`, `false`, or `no`, then Rerun is disabled.
+    inline void set_default_enabled(bool default_enabled) {
+        RerunGlobalConfig::instance().default_enabled.store(
+            default_enabled,
+            std::memory_order_seq_cst
+        );
+    }
+
+    /// Check if Rerun is enabled.
+    inline bool is_default_enabled() {
+        return RerunGlobalConfig::instance().default_enabled.load(std::memory_order_relaxed);
```

Comment:
Honestly, this being C++, I'd just make it all `seq_cst`... one less thing to worry about and it's not like we're fighting for contention here.

Reply:
I don't disagree with the sentiment, but seq_cst can be quite expensive on arm though and we want to do this thing in a lot of places.
If another thread changes the enabled state there's no way anyways to determine when that happens in relation to instruction on the other state. Afaik the only thing that this protects us from is that instructions that in some cases the code "look like they come later" may see a different enabled state than those that "look like they come earlier" (as long as it doesn't break logic on a single thread).

Reply:
> we want to do this thing in a lot of places.

That doesn't sound right? `is_default_enabled` should only be relevant when creating a recording stream, which is quite rare.

Reply:
Yeah, again, the usage of atomic here stems from when we thought this check was going to happen inside every log call as opposed to just during stream construction.